sktime/forecasting/base/convertIO/_convertIO.py

```diff
@@ -0,0 +1,190 @@
+# -*- coding: utf-8 -*-
+"""
+Machine type converters for scitypes
+
+Exports
+-------
+convert : dictionary of functions, with signature as below
+        indexed by triples of types
+         1st element = convert from - type
+         2nd element = convert to - type
+         3rd element = considered as this scitype - string
+        elements are conversion functions of machine type (1st) -> 2nd
+
+convert_to : function/dispatch version of convert, with 1st elt inferred
+
+
+-----------------------------------------
+
+Function signature of
+convert[(from_type, to_type, as_scitype)]
+
+Parameters
+----------
+what : from_type - object to convert
+store : reference of storage for lossy conversions, default=None (no store)
+
+Returns
+-------
+converted_what : to_type - object what converted to to_type
+
+Raises
+------
+ValueError and TypeError, if requested conversion is not possible
+                            (depending on conversion logic)
+
+-----------------------------------------
+
+Function signature of convert_to
+
+Parameters
+----------
+what : object to convert (any type)
+to_type : type, the type to convert "what" to
+as_scitype : str - name of scitype the object "what" is considered as
+store : reference of storage for lossy conversions, default=None (no store)
+
+Returns
+-------
+converted_what : to_type - object what converted to to_type
+
+Raises
+------
+TypeError if no suitable key in convert exists
+    i.e., (type(what), to_type, as_scitype)
+
+
+copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""
+
+
+__author__ = ["fkiraly"]
+
+
+import numpy as np
+import pandas as pd
+
+
+# dictionary indexed by triples of types
+#  1st element = convert from - type
+#  2nd element = convert to - type
+#  3rd element = considered as this scitype - string
+# elements are conversion functions of machine type (1st) -> 2nd
+
+convert = dict()
+
+
+def convert_identity(what, store=None):
+
+    return what
+
+
+# assign identity function to type conversion to self
+for tp in [pd.Series, pd.DataFrame, np.array]:
+    convert[(tp, tp, "Series")] = convert_identity
+
+
+def convert_UvS_to_MvS_as_Series(what: pd.Series, store=None) -> pd.DataFrame:
+
+    if not isinstance(what, pd.Series):
+        raise TypeError("input must be a pd.Series")
+
+    if isinstance(store, dict) and "cols" in store.keys and len(store["cols"]) == 1:
```

Comment:
* checking that the length of columns is 1 might be redundant? in what situation would this particular convert function be called and the checking of length be important?
* secondly, if length of stored columns is greater than 1, then it suggests something has gone wrong somewhere (I think), and maybe we should raise an error?

Reply:
>in what situation would this particular convert function be called and the checking of length be important?

when the columns in the store, for some reason, have the wrong length, the constructor with `columns=` throws an error